WooCommerce/UITestsFoundation/Screens/Products/ProductSearchScreen.swift

```diff
@@ -0,0 +1,47 @@
+import ScreenObject
+import XCTest
+
+public final class ProductSearchScreen: ScreenObject {
+
+    private let searchTextFieldGetter: (XCUIApplication) -> XCUIElement = {
+        $0.descendants(matching: .other).element(matching: .any, identifier: "product-search-screen-search-field")
+    }
+
+    private var searchTextField: XCUIElement { searchTextFieldGetter(app) }
+    private var updatedSearchResultsID = "updated-search-results-table-view"
+    public static let defaultSearchResultsID = "default-search-results-table-view"
+
+    public init(app: XCUIApplication = XCUIApplication()) throws {
+        try super.init(
+            expectedElementGetters: [ searchTextFieldGetter ],
+            app: app
+        )
+    }
+
+    public func enterSearchCriteria(text: String) throws -> Self {
+        searchTextField.tap()
+        searchTextField.enterText(text: text)
+        return self
+    }
+
+    @discardableResult
+    public func verifyNumberOfProductsOnSearchResults(expectedNumberOfProducts: Int, accessibilityId: String? = defaultSearchResultsID) throws -> Self {
```

Comment:
Since the function is named `verifyNumberOfProductsOnSearchResults` (just a nitpick: I think it should be `In`, not `On` 🤔), it's intuitive that we're verifying the products number, so I don't think we should repeat that in the argument name (also, you can have an external name for the argument, to have the calls more "human-like"):

```suggestion
    public func verifyNumberOfProductsInSearchResults(equals expectedNumber: Int, accessibilityId: String? = defaultSearchResultsID) throws -> Self {
```

Then the call will look like:

```swift
try verifyNumberOfProductsInSearchResults(equals: 1, accessibilityId: updatedSearchResultsID)
```

Reply:
Oh nice! Using the external name `equals` does make it easier to read. Updated in https://github.com/woocommerce/woocommerce-ios/pull/9276/commits/68fe837e960c4576b922f222055c502f798806bf